datalad/distribution/get.py

```diff
@@ -503,53 +509,84 @@ def _install_necessary_subdatasets(
+            if sub.get('state', None) != 'absent':
+                rec = get_status_dict('install', ds=subds, status='notneeded', logger=lgr, refds=refds_path)
+                subs_notneeded.append(rec)
+                yield rec
+                # do not continue, even if an intermediate dataset exists it
+                # does not imply that everything below it does too
+            else:
+                # TODO: here we need another "ds"!  is it within "sub"?
+                yield from _install_subds_from_flexible_source(
+                    Dataset(ds_path), sub, reckless=reckless, description=description)
+
+            if not subds.is_installed():
+                # an error result was emitted, and the external consumer can decide
+                # what to do with it, but there is no point in recursing into
+                # something that should be there, but isn't
+                lgr.debug('Subdataset %s could not be installed, skipped', subds)
+                return
+
+            # recurse
+            # we can skip the start expression, we know we are within
+            for res in _recursive_install_subds_underneath(
+                    subds,
+                    recursion_limit=recursion_limit - 1 if isinstance(recursion_limit, int) else recursion_limit,
                     reckless=reckless,
-                    description=description):
-                # yield everything to let the caller decide how to deal with
-                # errors
-                yield res
-        if not subds.is_installed():
-            # an error result was emitted, and the external consumer can decide
-            # what to do with it, but there is no point in recursing into
-            # something that should be there, but isn't
-            lgr.debug('Subdataset %s could not be installed, skipped', subds)
-            continue
-        # recurse
-        # we can skip the start expression, we know we are within
-        for res in _recursive_install_subds_underneath(
-                subds,
-                recursion_limit=recursion_limit - 1 if isinstance(recursion_limit, int) else recursion_limit,
-                reckless=reckless,
-                refds_path=refds_path):
-            yield res
+                    refds_path=refds_path,
+                    jobs=jobs,
+                    producer_only=True  # we will be adding to producer queue
+            ):
+                producer_consumer.add_to_producer_queue(res)
+
+        producer = gen_subs_to_install()
+        if producer_only:
+            yield from producer
+        else:
+            producer_consumer = ProducerConsumerProgressLog(
+                producer,
+                consumer,
+                # no safe_to_consume= is needed since we are doing only at a single level ATM
+                label="Installing",
+                unit="datasets",
+                jobs=jobs,
+                lgr=lgr
+            )
+            yield from producer_consumer
+
```

Comment:
The entire function diff looks weird. I think it is indented one level to much. I am blind?

Reply:
yeah `get` is the trickiest case -- consumer needs to feed producer (as we install and discover more of new subdatasets to be installed).  Even though I have tried to keep diff at minimum, it still was largely code movement, but also (re)indentation, so the diff could be odd.

Reply:
but hey -- you are right!!! Probably in my striving for minimal diff I over indented the function text body! ;) I will fix that -- thanks for spotting!

Reply:
I de-indented that function locally.  I did not push to not torture CI - since may be more changes to come